app/models/event.rb

```diff
@@ -14,4 +14,8 @@ class Event < ApplicationRecord
   validates :address,     presence: true
 
   scope :not_happened, -> { where('occurred_at > ?', Time.current.yesterday).order(occurred_at: :asc) }
+
+  def decorate(view_context)
```

Comment:
I think if we change the controller the way you suggested below, we can get rid of this entirely.